x-pack/legacy/plugins/security/server/routes/api/v1/authenticate.js

```diff
@@ -82,6 +83,36 @@ export function initAuthenticateApi({ authc: { login, logout }, config }, server
     }
   });
 
+  /**
+   * The route should be configured as a redirect URI in OP when OpenID Connect implicit flow
+   * is used, so that we can extract authentication response from URL fragment and send it to
+   * the `/api/security/v1/oidc` route.
+   */
+  server.route({
+    method: 'GET',
+    path: '/api/security/v1/oidc/implicit',
+    config: { auth: false },
+    async handler(request, h) {
+      const legacyConfig = server.config();
+      const basePath = legacyConfig.get('server.basePath');
+
+      const nonce = await generateCSPNonce();
+      const cspRulesHeader = createCSPRuleString(legacyConfig.get('csp.rules'), nonce);
+      return h.response(`
```

Comment:
> This is likely a naive question.

And what is the question? :slightly_smiling_face: So, I assume you're asking why we can't just say OP and RP (ES) to use `response_mode=query` for the implicit flow. Per [OAuth 2.0 Multiple Response Type Encoding Practices](https://openid.net/specs/oauth-v2-multiple-response-types-1_0.html#id_token) it seems that query is forbidden for `id_token` and `token` response types:

> 	The default Response Mode for this Response Type is the fragment encoding and the query encoding MUST NOT be used.

cc @jkakavas

Reply:
Okta seems to obey this rule too https://developer.okta.com/docs/reference/api/oidc/#request-parameters

Reply:
Ah, thanks for the clarification. I stopped reading the specs too soon.

Reply:
fragment is used on purpose so that the access token and the id token are not sent as part of any requests to a back-end server ( The assumption was that the implicit flow would be best used by javascript clients running in the browser ) and thus don't get leaked in logs or be available more than needed on the wire. We (and many other clients) have the validation on the server side so we end up making a request with it to the kibana server eitherway so there goes that.. In any case as @azasypkin mentioned we're not allowed to use anything else for `response_type`

Reply:
Thanks @jkakavas